presto-hive/src/main/java/com/facebook/presto/hive/pagefile/PageFileWriterFactory.java

```diff
@@ -104,6 +111,43 @@ public static void createEmptyPageFile(
         }
     }
 
+    public static HiveCompressionCodec getCompressionCodec(String compression)
+    {
+        try {
+            return HiveCompressionCodec.valueOf(compression);
+        }
+        catch (IllegalArgumentException | NullPointerException e) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, format("Unrecognized compression name %s", compression), e);
+        }
+    }
+
+    public static PagesSerde createPagesSerdeForPageFile(BlockEncodingSerde blockEncodingSerde, HiveCompressionCodec compressionCodec)
+    {
+        if (!compressionCodec.isSupportedStorageFormat(PAGEFILE)) {
+            throw new PrestoException(
+                    GENERIC_USER_ERROR,
+                    format("%s compression is not supported for %s", compressionCodec.name(), PAGEFILE.getOutputFormat()));
+        }
+
+        PageCompressor pageCompressor = null;
+        PageDecompressor pageDecompressor = null;
+
+        switch (compressionCodec) {
+            case NONE:
+                break;
+            case SNAPPY:
+                pageCompressor = new BasicPageCompressor(new SnappyCompressor());
+                pageDecompressor = new BasicPageDecompressor(new SnappyDecompressor());
+                break;
+            default:
```

Comment:
Throwing an error here caused test failures: "om.facebook.presto.spi.PrestoException: GZIP compression is not supported with PAGEFILE"   because these tests use "GZIP" as default file compression method.  For unsupported `HiveCompressionCodec`,  should the default behaviour be no compression rather than throwing an error?

Reply:
This seems a good solution to not changing the default setting for `hive.compression-codec`

Reply:
May be you should have a separate compression config for pageFileStorage, that way you can have your own default. It would also allow you to configure on clusters without touching compression for other storage.

Reply:
> May be you should have a separate compression config for pageFileStorage, that way you can have your own default. It would also allow you to configure on clusters without touching compression for other storage.

cc @arhimondr This is also a good solution. Is there a need to support GZIP for page file?

Reply:
In practice the pagefile format is going to be used only for temporary tables. There's a configuration property for temporary tables already: https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java#L158. Exploding properties is generally not desirable, as it makes the configuration process not straightforward. Unless there's a clear justification for adding more properties I would suggest to restrain from it.

@viczhang861 If you think that implementing GZIP will take too much time feel free not to do this. I just thought that it might be a simple change.